internal/cmd/rootcmd/signal.go

```diff
@@ -0,0 +1,54 @@
+package rootcmd
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/bombsimon/logrusr/v2"
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/manager"
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/util"
+)
+
+var shutdownSignals = []os.Signal{os.Interrupt, syscall.SIGTERM}
+
+var onlyOneSignalHandler = make(chan struct{})
```

Comment:
The value of this variable is to allow us to panic to avoid situations where two signal handlers have been run, the context is that this will basically only happen if we make a mistake in the code. In it's current state, the panic message that occurs if this happens is not clear about this, it will simply inform the user that they've closed a closed channel.

I think we should re-work this implementation so that we can provide a clear message about the problem that was encountered, and preferably exit with that error instead of panicking. I'm open to your preference on how this is done, for instance you could handle it with a `recover()` or by replacing the channel with a `sync.Mutex` and doing a `TryLock()` which results in the error if the mutex is already locked.